play-framework/websockets/app/actors/Messenger.java

```diff
@@ -0,0 +1,114 @@
+import com.fasterxml.jackson.databind.JsonNode;
+import dto.MessageDTO;
+import dto.RequestDTO;
+import utils.MessageConverter;
+
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Created by alfred on 02 Mar 2019
+ */
+public class Messenger extends AbstractActor {
+    private final LoggingAdapter log = Logging.getLogger(getContext().getSystem(), this);
+
+    private final ActorRef out;
+
+    public Messenger(ActorRef out) {
+        this.out = out;
+    }
+
+    public static Props props(ActorRef out) {
+        return Props.create(Messenger.class, () -> new Messenger(out));
+    }
+
+    @Override
+    public void preStart() throws Exception {
+        log.info("Messenger actor started at {}",
+          OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME));
+    }
+
+    @Override
+    public void postStop() throws Exception {
+        log.info("Messenger actor stopped at {}",
+          OffsetDateTime.now().format(DateTimeFormatter.ISO_OFFSET_DATE_TIME));
+    }
+
+    private void onSendMessage(JsonNode jsonNode) {
+        final RequestDTO requestDTO = MessageConverter.jsonNodeToRequest(jsonNode);
+        final String message = requestDTO.getMessage().toLowerCase();
+        if("stop".equals(message)) {
+            MessageDTO messageDTO = createMessageDTO("1", "1", "Stop", "Stopping actor");
+            out.tell(MessageConverter.messageToJsonNode(messageDTO), getSelf());
+            self().tell(PoisonPill.getInstance(), getSelf());
+        } else {
+            log.info("Actor received. {}", requestDTO);
+            processMessage(requestDTO);
+        }
+    }
+
+    private MessageDTO createMessageDTO(String userId, String id, String title, String message) {
+        MessageDTO messageDTO = new MessageDTO();
+        messageDTO.setUserId(UUID.randomUUID().toString());
+        messageDTO.setId(UUID.randomUUID().toString());
+        messageDTO.setTitle("Self Kill");
+        messageDTO.setBody("Stopping actor");
+        return messageDTO;
+    }
+
+    private void processMessage(RequestDTO requestDTO) {
+        CompletionStage<HttpResponse> responseFuture = getRandomMessage();
+        responseFuture.thenCompose(this::consumeHttpResponse)
+          .thenAccept(messageDTO ->
+            out.tell(MessageConverter.messageToJsonNode(messageDTO), getSelf()));
+    }
+
+    private CompletionStage<HttpResponse> getRandomMessage() {
+        int postId = ThreadLocalRandom.current().nextInt(0, 100);
+        return Http.get(getContext().getSystem()).singleRequest(
+          HttpRequest.create("https://jsonplaceholder.typicode.com/posts/" + postId)
+        );
+    }
+
+    private void discardEntity(HttpResponse httpResponse, Materializer materializer) {
+        final HttpMessage.DiscardedEntity discarded = httpResponse.discardEntityBytes(materializer);
```

Comment:
Do we need the `discarded` temp variable ? Can we not just call fluently from `httpResponse.discardEntityBytes(...)`?

Reply:
Fixed.